tools/python_api/src_py/query_result.py

```diff
@@ -23,6 +23,12 @@ def __init__(self, connection, query_result):
         self._query_result = query_result
         self.is_closed = False
 
+    def __enter__(self):
```

Comment:
@alexander-beedie Switching to the context manager protocol would count as a breaking change, correct? As in, the way it's done now `conn.execute(...)` would fail and the user is required to specify `with conn.execute` regardless of whether or not Polars is the desired output format?

Reply:
Fortunately not; it's completely optional 😉 If you don't use it as a context manager then there's no change; you get identical behaviour to what happens now.